src/storage/src/hummock/local_version/local_version_impl.rs

```diff
@@ -537,17 +539,13 @@ impl LocalVersion {
                         "an commit_epoch call should always insert sst into L0, but not insert to {}",
                         insert_sst_level_id
                     );
-                    if let Some(ssts) = compaction_group_ssts.remove(compaction_group_id) {
-                        assert!(
-                            check_subset_preserve_order(
-                                ssts.iter().map(|info| info.id),
-                                insert_table_infos.iter().map(|info| info.id)
-                            ),
-                            "order of local synced ssts is not preserved in the global inserted sst. local ssts: {:?}, global: {:?}",
-                            ssts.iter().map(|info| info.id).collect_vec(),
-                            insert_table_infos.iter().map(|info| info.id).collect_vec()
+                    insert_table_infos.retain(|info| sst_ids.contains(&info.id));
+                    if !insert_table_infos.is_empty() {
```

Comment:
> Why removing `check_subset_preserve_order` check? I think we can still keep it. cc @wenym1

because some of SSTs in this set would be migrated into other compaction groups by meta